src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -374,6 +480,29 @@ public FilterAndTransform(string filterAndPayloadSpec, int startIdx, int endIdx,
                     }
                 }
 
+                Action<string, string, IEnumerable<KeyValuePair<string, string>>> writeEvent = null;
+                if (activityName != null && activityName.Contains("Activity"))
+                {
+                    MethodInfo writeEventMethodInfo = typeof(DiagnosticSourceEventSource).GetTypeInfo().GetDeclaredMethod(activityName);
+                    if (writeEventMethodInfo != null)
+                    {
+                        // This looks up the activityName (which needs to be a name of an event on DiagnosticSourceEventSource
+                        // like Activity1Start and returns that method).   This allows us to have a number of them and this code
+                        // just works.  
+                        try
+                        {
+                            writeEvent = (Action<string, string, IEnumerable<KeyValuePair<string, string>>>)
+                                writeEventMethodInfo.CreateDelegate(typeof(Action<string, string, IEnumerable<KeyValuePair<string, string>>>), _eventSource);
+                        }
+                        catch (Exception) { }
```

Comment:
Given that now catching 'Exception' means 'Catch non-corrupting exceptions', I think it is OK.   The LOGIC of the code makes sense.  If the body fails (for ANY reason), we can keep going, and we would want to (with an appropriate error messsage).   Trying to catch anything more specific leaves open the possibility that you will crash the PROGRAM (you consider it fatal).   We really don't like that in logging (we would prefer logging to fail and let the app run).   

Now it is true I am probably not uniformly careful about never causing an exception, but in this code it is likely based on USER INPUT, and the reflection system is complex enough that it may fail with different errors that I am not anticipating, so a more general catch seems reasonable.